dht/i_partitioner.cc

```diff
@@ -179,6 +179,20 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_not_empty(const dht::token_range& r) {
+    bool is_bounded = r.start() && !r.start()->value().is_minimum() && r.end() && !r.end()->value().is_maximum();
+    if (is_bounded) {
+        uint64_t start_raw = r.start()->value().raw();
+        uint64_t end_raw = r.end()->value().raw();
+        return end_raw - start_raw >= (!r.start()->is_inclusive() + !r.end()->is_inclusive());
+    } else {
+        const static dht::token_range full_range = dht::token_range::make(
+                range_bound<dht::token>(token::from_int64(std::numeric_limits<int64_t>::min() + 1), true),
```

Comment:
Maybe use range_bound<dht::token>(minimum_token(), false)

Reply:
We want (-inf, int64min) to be empty, but your bound would result in a (-inf, before_all_keys) intersection.

Reply:
Ok.